feat: Refactor the VS code extension page DOCS-596 (#1863)
codacy/docs: docs/getting-started/integrating-codacy-with-visual-studio-code.md

In the Problems tab, Codacy displays the same Quality issues you find in the main view and lets you navigate to the exact line of code where the issue was found.

!!! note
    Code coverage, duplicates, and complexity aren't currently shown in the Problems tab.

To review Quality issues:

1.  Open the **Problems tab** (use `Ctrl+Shift+M` on Windows/Linux or `Cmd+Shift+M` on macOS).

1.  Click the name of the issue you want to review.

    ![Navigate to a specific issue from the Problems tab](images/codacy-vscode-extension-problems-tab.png)

1.  Hover over a highlighted issue in the code editor to view available actions and suggested quick fixes (if available).

    For a list of tools that support quick fixes, see [Supported languages and tools](./supported-languages-and-tools.md#supported-languages-and-tools).

1.  Once you've addressed the problems in your code, push your changes to the Git provider so that Codacy analyzes the updated code.

When the analysis is complete, the Codacy extension automatically refreshes the pull request analysis result. You can also refresh the pull request data manually by clicking the Refresh Pull Request button in the main view.

## See also

> -   [Troubleshooting the Codacy VS Code extension](https://marketplace.visualstudio.com/items?itemName=codacy-app.codacy#troubleshooting)

Reviewer: @nicklem Don't we want to link to the [GitHub project](https://github.com/codacy/codacy-vscode-extension#troubleshooting) instead? 🤔

Author: That's also an option. It depends if we want to keep the users on the front-facing pages vs. the Git repository. We can update this in a follow-up.

1.  Install the extension from the [Visual Studio Marketplace](https://marketplace.visualstudio.com/items?itemName=codacy-app.codacy) or through the [Extensions view in VS Code](https://code.visualstudio.com/docs/editor/extension-marketplace#_browse-for-extensions).

    Alternatively, you can install it manually by [downloading the latest release as a VSIX package](https://github.com/codacy/codacy-vscode-extension/releases).

## Getting pull request quality and coverage data

To see Codacy quality and coverage data for the pull request you're working on, follow these steps:

1.  Open the repository directory in VS Code.

    !!! note
        If the repository isn't on Codacy yet, [add it to Codacy](../organizations/managing-repositories.md#adding-a-repository) first.

1.  Open the main view by clicking the **Codacy logo** in the activity bar or the **Codacy tab** in the status bar.

    ![Codacy main view](images/codacy-vscode-extension-sign-in.png)

1.  If you’re not signed in, click the **Sign in** button to authorize VS Code on Codacy.

> 1.  Check out the branch you're working on.

Reviewer: Maybe we can avoid the admonition by including the information within the step.

Also, I believe it makes sense to move this step to step 2 of the procedure, right after opening the repository directory in VS Code and before opening the extension. 🤔 

Thoughts?

```suggestion
1.  Check out the branch associated with the open pull request you're working on.
```

Author: Sounds good! [Done](https://github.com/codacy/docs/pull/1863/commits/ef6dea95b1daaa2883400cada8f0cda27ca99638) 👍

# Integrating Codacy with Visual Studio Code

!!! info "To use this extension you need a [Codacy account](https://www.codacy.com/signup-codacy)"

The Codacy extension for VS Code displays information about the quality of a pull request, highlights problematic code patterns, and displays code quality metrics directly in VS Code.

Use this extension to get the full list of problems found by Codacy for the pull request you’re working on and to navigate to any Quality issue that you want to review and fix.

## Installing the Codacy VS Code extension

1.  Make sure that the repository you’re working on is analyzed by Codacy and that you have at least [Repository Read permissions](../organizations/roles-and-permissions-for-organizations.md).

    !!! tip
        If this is your first time using Codacy, see [how to add and analyze your first repository](./codacy-quickstart.md#adding-your-first-repository).

> 1.  Install the extension from the [Visual Studio Marketplace](https://marketplace.visualstudio.com/items?itemName=codacy-app.codacy) or through the Extensions view in VS Code.

Author: I was a bit on the fence and opted to remove it, but since you're mentioning it, too, let's keep it for now.

    Alternatively, you can install it manually by [downloading the latest release as a VSIX package](https://github.com/codacy/codacy-vscode-extension/releases).

## Getting pull request quality and coverage data

To see Codacy quality and coverage data for the pull request you're working on, follow these steps:

1.  Open the repository directory in VS Code.

    !!! note
        If the repository isn't on Codacy yet, [add it to Codacy](../organizations/managing-repositories.md#adding-a-repository) first.

> 1.  Open the main view by clicking the **Codacy logo** in the status bar or the **Codacy tab** in the activity bar.

Author: Good catch!

    ![Codacy main view](images/codacy-vscode-extension-sign-in.png)

1.  If you’re not signed in, click the **Sign In** button to authorize VS Code on Codacy.

1.  Check out the branch you're working on.

    !!! note
        The branch must be associated with an open pull request analyzed by Codacy.

> After completing these steps, Codacy displays a summary of all problems for the pull request in the main view and a list of found Quality issues in VS Code's Problems tab.

Reviewer: Breaking the sentence for readability.
Also, maybe we can avoid "problems" by using "result of Codacy analysis"? (I heard this expression in the demo)

```suggestion
After completing these steps, the main view shows the result of the Codacy analysis for the pull request. The VS Code's Problems tab lists the Quality issues found.
```

Author: Looks good! Just a nitpick, removing the possessive `s`.

```suggestion
After completing these steps, the main view shows the result of the Codacy analysis for the pull request. The VS Code Problems tab lists the Quality issues found.
```

![Codacy main view](images/codacy-vscode-extension-main-view.png)

> The main view displays the following:

Reviewer: Do you think that we should mention at this point that the information displayed here is the same as displayed on the [Codacy UI](https://docs-596-vs-code-documentation-refa--docs-codacy.netlify.app/repositories/pull-requests/)? 🤔 

```suggestion
The main view displays the information about the code quality changes introduced by the pull request:
```

    The number next to each file name is the total number of Quality issues that the pull request adds to or removes from that file. The number farther to the right is the number of new Medium or Critical Quality issues. If there are any such issues, the file name is also highlighted in yellow (Medium) or red (Critical).

    ![Important issues](./images/codacy-vscode-extension-important-issues.png)

-   The **diff coverage and coverage variation** introduced by the pull request. These are the same values you find in the [Pull request coverage overview](../repositories-coverage/pull-requests.md#coverage-overview) panel in the Codacy app.

    The percentage next to each file name is the coverage variation for that file.

-   Sequences of **duplicate code** (clones). These are the same ones you find in the [Quality Duplication tabs](../repositories/pull-requests.md#duplication-tabs) in the Codacy app.

-   Variations in **code complexity**.

## Reviewing pull request issues

In the Problems tab, Codacy displays the same Quality issues you find in the main view and lets you navigate to the exact line of code where the issue was found.

!!! note

>     Code coverage, duplicates, and complexity aren't currently shown in the Problems tab.

Reviewer: Avoid passive voice.

```suggestion
    The Problems tab doesn't list code coverage, duplicates, and complexity.
```

Author: Here it's intentional. I'm using the passive to start the sentence with the information of interest (a list of excluded categories). Thoughts?

To review Quality issues:

1.  Open the **Problems tab** (use `Ctrl+Shift+M` on Windows/Linux or `Cmd+Shift+M` on macOS).

1.  Click the name of the issue you want to review.

    ![Navigate to a specific issue from the Problems tab](images/codacy-vscode-extension-problems-tab.png)

1.  Hover over a highlighted issue in the code editor to view available actions and suggested quick fixes (if available).

>     For a list of tools that support quick fixes, see [Supported languages and tools](./supported-languages-and-tools.md#supported-languages-and-tools)

Reviewer: Missing punctuation.

```suggestion
    For a list of tools that support quick fixes, see [Supported languages and tools](./supported-languages-and-tools.md#supported-languages-and-tools).
```

> 1.  Once you've addressed the problems in your code, push your changes to the Git provider so that Codacy analyzes the updated code. Once the analysis is complete, the Codacy extension automatically refreshes the pull request data.

Reviewer: Suggestion to avoid having two "once". I think the second part can be put out of the procedure, as something that it's triggered after you execute the last step (push changes).
WDYT?

```suggestion
1.  Once you've addressed the problems in your code, push your changes to the Git provider so that Codacy analyzes the updated code.

When the analysis is complete, the Codacy extension automatically refreshes the pull request analysis result.
```

    You can also refresh the pull request data manually by clicking the Refresh Pull Request button in the main view.

> ## Troubleshooting

Reviewer: We could consider moving this Troubleshooting section to the [extension documentation](https://github.com/codacy/codacy-vscode-extension), leaving only the "happy path" in the Codacy product documentation. We can then link it from here as See Also.
WDYT?
